freeze/just_wrapper

```diff
@@ -0,0 +1,58 @@
+#!/usr/bin/env bash
+
+set -eu
+
+# Check the location of the just executable, then CWD for local.env files, and
+# load them in that order.
+for check_dir in "${ARCHIVE_DIR}" "${USER_PWD}"; do
+  if [ -f "${check_dir}/local.env" ]; then
+    set -a
+    source "${check_dir}/local.env"
+    set +a
+  fi
+done
+
+if [ -n "${VSI_COMMON_DIR+set}" ] && [ ! -d "${VSI_COMMON_DIR}" ]; then
+  echo "ERROR: the environment variable VSI_COMMON_DIR is set to a directory that does not exist;"
+  echo "       please either unset or correct it"
+  exit 1
+fi
+
+if [ -z ${VSI_COMMON_DIR+set} ]; then
+  VSI_COMMON_DIR="$(dirname "${BASH_SOURCE[0]}")/.."
+fi
+source "${VSI_COMMON_DIR}/env.bsh"
+
+# Just Run On Deployment EnvirOnment
+export JUST_RODEO=1
+
+# This came from juste, so probably right
+if [ "${OS-}" = "Windows_NT" ]; then
+  if [ "${JUST_FROZEN-}" = "1" ] || [ "${JUST_IS_TTY-}" = "1" ]; then
+    export JUST_FROZEN=1
+    exec just ${@+"${@}"}
+  else
+    export JUST_FROZEN=1
+
+    stdin_name="$(powershell -NoProfile -InputFormat None -ExecutionPolicy Bypass -Command "${VSI_COMMON_DIR}/windows/get_stdin_name.ps1")"
+
+    if [ "${stdin_name}" = "None" ] || [[ ${stdin_name} != *-from-master ]]; then
+      # Even though a pipe isn't a TTY, I don't need a TTY, to mock the system
```

Comment:
What is this saying

Reply:
"so mock the system"
A pipe doesn't have a TTY, but I don't need a TTY, so I mock the TTY flag so just doesn't allocate a TTY (via powershell) It's a windows implementation detail.